### DISCUSSION TITLE
Use GoRules without Graphic interface

### DISCUSSION THREAD

**Discussion by tlaraMQ** [gorules/zen#130] [Q&A]
Hello, 
My organization is looking for a Rules Engine. I think GoRules is an interesting option because of its functionalities and the possibility of using it with Python and Rust. However, we'd like to use the Rules Engine on our website. Is there a way to define the decision graph without the graphical interface? Using a JSON format seems useful for our use case, but I'd like to ask if some documentation indicates how to define this structure without translating it using a graphic interface.

Thanks!

**Comment by ivanmiletic** [comment]
Hi @tlaraMQ , we have plans to expose Zen Engine directly to Web over WASM. In this case you would be editing the rules over UI and pulling JSON and evaluating directly on the Web - in plans for the end of the year. As for the standard, we also have a plan to prepare documentation for the standard.

Both of those are not a priority as we have 2 big features coming down the road over next couple of months.

**Comment by tlaraMQ** [comment]
Hi @ivanmiletic, thank you for your answer. Sorry if I didn't express my question properly. 

As far as I understand (please correct me if I'm wrong). The use of GoRules using a programming language (Python, Go, Rust) consists of two parts. First, we have the definition of the decision graph, which is then exported into a JSON file; and second, we have the import of JSON and the execution using the programming language on some kind of local machine. 

We would like to use a Rules Engine on a product that has a decision module. We interact with the module, specifically with the rules definition, using a website with a previously defined set of variables and conditions. Is there a way to define the JSON file (and hence, the graph structure) without using the graphical interface? Is there any documentation on it?
